### PR TITLE
resolve issue #676

### PR DESCRIPTION
**Pull request by nmccready**
- on creation copy the model so changes to lObjects do not affect the model/trigger scope digest 
- model differences working now where copy in _addMarker is very important

### CODE REVIEW

**Comment by nmccready** [comment]
@zacronos @tombatossals

File: src/directives/markers.js [hunk 1]
```diff
@@ -61,7 +61,11 @@ angular.module("leaflet-directive").directive('markers',
             }
 
             if (!isDefined(leafletMarkers[newName])) {
-                var model = markersToRender[newName];
+                //(nmccready) very important to not have model changes when lObject is changed
+                //this might be desirable in some cases but it causes two-way binding to lObject which is not ideal
+                //if it is left as the reference then all changes from oldModel vs newModel are ignored
+                //see _destroy (where modelDiff becomes meaningless if we do not copy here)
+                var model = Helpers.copy(markersToRender[newName]);
```
Comment (nmccready) [inline src/directives/markers.js]: This logic should be in all directives.

File: src/directives/markers.js [hunk 2]
```diff
@@ -111,16 +115,27 @@ angular.module("leaflet-directive").directive('markers',
...
                 $log.debug(errorHeader + "[markers] destroy: ");
                 hasLogged = true;
             }
-            if (!isDefined(markerModels) || !Object.keys(markerModels).length ||
-                !isDefined(markerModels[name]) || !Object.keys(markerModels[name]).length) {
+
+            if(doCheckOldModel){
+              //might want to make the option (in watch options) to disable deep checking
+              //ie the options to only check !== (reference check) instead of angular.equals (slow)
+              modelIsDiff = !angular.equals(markerModels[name],oldMarkerModels[name]);
```
Comment (nmccready) [inline src/directives/markers.js]: modelDiff added to check each model against an old one if it exists.

**Comment by zacronos** [comment]
LGTM

**Pull request merged**